api/v1beta1/nodepool_types.go

```diff
@@ -185,6 +185,24 @@ const (
 	UpgradeTypeInPlace = UpgradeType("InPlace")
 )
 
+func (p *UpgradeType) String() string {
+	return string(*p)
+}
+
+func (p *UpgradeType) Set(s string) error {
+	switch s {
+	case string(UpgradeTypeReplace), string(UpgradeTypeInPlace):
```

Comment:
Suggestion: I would change this up a bit to accept case insensitivity for the validation check but set the value to the right case. Right now, if you enter `replace` or `inplace`, you'll get an error. 

```
brcox@brcox-mac hypershift % ./bin/hypershift create cluster aws --name $CLUSTER_NAME --node-pool-replicas=$NODEPOOL_REPLICAS --base-domain $BASE_DOMAIN --pull-secret $PULL_SECRET --aws-creds $AWS_CREDS --region $REGION --render --upgrade-type replace
Error: invalid argument "replace" for "--upgrade-type" flag: unknown upgrade type used replace

brcox@brcox-mac hypershift % ./bin/hypershift create cluster aws --name $CLUSTER_NAME --node-pool-replicas=$NODEPOOL_REPLICAS --base-domain $BASE_DOMAIN --pull-secret $PULL_SECRET --aws-creds $AWS_CREDS --region $REGION --render --upgrade-type inplace
Error: invalid argument "inplace" for "--upgrade-type" flag: unknown upgrade type used inplace
```

It's a better UX if I can not worry about the case sensitivity and the code can set the right case for me.